heartbeat/podman-etcd

```diff
@@ -1119,6 +1245,58 @@ ensure_pod_manifest_exists()
 	return "$OCF_SUCCESS"
 }
 
+manage_etcd_pod_manifest_for_reuse() {
+	local cp_rc
+	local jq_filter
+	local diff_rc
+
+	# If the manifest copy doesn't exist, create it and ensure a new container.
+	if [ ! -f "$OCF_RESKEY_pod_manifest_copy" ]; then
+		if ! ocf_run cp -p "$OCF_RESKEY_pod_manifest" "$OCF_RESKEY_pod_manifest_copy"; then
+			cp_rc="$?"
+			ocf_log err "Could not create a working copy of $OCF_RESKEY_pod_manifest, rc: $cp_rc"
+			return "$OCF_ERR_GENERIC"
+		fi
+		ocf_log info "A working copy of $OCF_RESKEY_pod_manifest was not found. Created a new copy now."
+		ocf_log info "A new etcd container will be created"
+		OCF_RESKEY_reuse=0
+		return "$OCF_SUCCESS"
+	fi
+
+	# Decide whether to reuse the existing container or create a new one based
+	# on etcd pod manifest changes explicitly ignoring the revision, as the content
+	# might be the same even if the revision number has changed.
+
+	# NOTE: the pod manifest contains some values referred to its REVISION that we want to ignore
+	jq_filter='del(.metadata.labels.revision) | .spec.containers[] |= ( .env |= map(select( .name != "ETCD_STATIC_POD_VERSION" ))) | .spec.volumes |= map( select( .name != "resource-dir" ))'
+
+	ocf_run diff -s <(jq "$jq_filter" "$OCF_RESKEY_pod_manifest") <(jq "$jq_filter" "$OCF_RESKEY_pod_manifest_copy")
```

Comment:
does this print out the diff so we can see it in the runtime log?

Reply:
yes